frontend/packages/core/src/stepper.tsx

```diff
@@ -0,0 +1,186 @@
+    font: "#FFFFFF",
+  },
+  failed: {
+    background: "#DB3615",
+    border: "#DB3615",
+    font: "#FFFFFF",
+  },
+};
+
+const StepIcon: React.FC<StepIconProps> = ({ index, variant }) => {
+  const color = stepIconVariants[variant || "pending"];
+  let Icon = <>{index}</>;
+  if (variant === "success") {
+    Icon = <CheckIcon font={color.font} fontSize="large" />;
+  } else if (variant === "failed") {
+    Icon = <ClearIcon font={color.font} fontSize="large" />;
+  }
+  return (
+    <Circle background={color.background} border={color.border}>
+      <DefaultIcon font={color.font}>{Icon}</DefaultIcon>
+    </Circle>
+  );
+};
+
+const StepLabel = styled(Grid)({
+  fontWeight: 500,
+  fontSize: "0.875rem",
+  lineHeight: "1.125rem",
+});
+
+export interface StepProps {
+  label: string;
+  error?: boolean;
+}
+
+const Step: React.FC<StepProps> = ({ children }) => <>{children}</>;
+
+export interface StepperProps {
+  children?: React.ReactElement<StepProps>[] | React.ReactElement<StepProps>;
+}
+
+const Stepper: React.FC<StepperProps> = ({ children }) => {
+  const [activeStep, setActiveStep] = React.useState(0);
+  const steps = React.Children.toArray(children) as React.ReactElement<StepProps>[];
+  const stepCount = steps.length - 1;
+
+  const handleNext = () => {
+    setActiveStep(prevActiveStep => prevActiveStep + 1);
+  };
+
+  const handleBack = () => {
+    setActiveStep(prevActiveStep => prevActiveStep - 1);
+  };
+
+  const handleReset = () => {
+    setActiveStep(0);
+  };
+
+  return (
+    <div>
+      <MuiStepper activeStep={activeStep + 1} connector={<StepConnector />}>
+        {React.Children.map(children, (step: any, idx: number) => {
+          const stepProps = {
+            index: idx + 1,
+            variant: "pending" as StepIconVariant,
+          };
+          if (idx === activeStep) {
+            stepProps.variant = "active";
+          } else if (idx < activeStep) {
+            stepProps.variant = step.props.error ? "failed" : "success";
+          }
+
+          return (
+            <MuiStep key={step.label} style={{ padding: "0" }}>
+              <StepLabelIcon icon={<StepIcon {...stepProps} />} />
+            </MuiStep>
+          );
+        })}
+      </MuiStepper>
+      <Grid style={{ padding: "0 1.5rem 1.5rem 1.5rem" }} container justify="space-between">
+        {React.Children.map(children, (step: any) => (
+          <StepLabel item>{step.props.label}</StepLabel>
+        ))}
+      </Grid>
+
+      <div>
+        {steps[activeStep]}
+        {activeStep === stepCount || steps[(activeStep || 1) - 1].props.error ? (
+          <Button onClick={handleReset} text="Reset" variant="neutral" />
+        ) : (
+          <div>
+            <Button
+              disabled={activeStep === 0}
+              onClick={handleBack}
+              text="Back"
+              variant="neutral"
+            />
+            <Button onClick={handleNext} text={activeStep === stepCount ? "Finish" : "Next"} />
```

Comment:
should the controls be a part of the story (actions) and not the component itself?

Reply:
I figured as we use this we can shape how to hook into the stepper. My idea at the moment was to have a `onSubmit` for each step that triggers a callback during the transition phase but I think plugging this into a real world use case will flush out if that makes sense.